kolena/_utils/batched_load.py

```diff
@@ -79,6 +82,18 @@ def upload_data_frame_chunk(df_chunk: pd.DataFrame, load_uuid: str) -> None:
 DFType = TypeVar("DFType", bound=LoadableDataFrame)
 
 
+def upload_smart_chunk_data_frame(df: pd.DataFrame, uuid: str, rows: int = 1000) -> None:
+    batch_size = (BATCH_LIMIT // get_preflight_export_size(df, rows)) * rows
```

Comment:
could we add a short circuit logic to just use the len(df) as batch_size when the memory usage of the df is small to handle [this comment](https://github.com/kolenaIO/kolena/pull/565/files#r1553990628)?